src/ApolloConsumer.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient from 'apollo-client';
+const invariant = require('invariant');
+
+export interface ApolloConsumerProps {
+  children: (client: ApolloClient<any>) => React.ReactElement<any>;
```

Comment:
I think the return should `React.ReactNode` - it is looser than `ReactElement` and we have no real concerns about the return type.

Reply:
Agreed, as soon as I update the type to `React.ReactNode` I get the typescript error: 

```
Type '(props: ApolloConsumerProps & { children?: ReactNode; }, context: any) => ReactNode' is not assignable to type 'StatelessComponent<ApolloConsumerProps>'.
  Type 'ReactNode' is not assignable to type 'ReactElement<any>'.
    Type 'string' is not assignable to type 'ReactElement<any>'.
```

It seems that StatelessComponent not accept a `ReactNode` as the return type. Any ideas on how to fix this?

Reply:
That's mistake then, it should match render's return type. I was thinking it would be node because 16 allows arrays.

Reply:
Should say my mistake - I'm mobile